chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -113,74 +107,204 @@ pub fn validate_prepared_transactions(
     )
 }
 
+pub enum ImplicitTransitionParams {
+    ApplyChunk(ApplyChunkBlockContext, ShardContext),
+    Resharding(AccountId, RetainMode, ShardUId),
+}
+
 struct StateWitnessBlockRange {
-    /// Blocks from the last new chunk (exclusive) to the parent block
-    /// (inclusive).
-    blocks_after_last_chunk: Vec<Block>,
+    /// Transition parameters **after** the last chunk, corresponding to all
+    /// state transitions from the last new chunk (exclusive) to the parent
+    /// block (inclusive).
+    implicit_transition_params: Vec<ImplicitTransitionParams>,
     /// Blocks from the last last new chunk (exclusive) to the last new chunk
```

Comment:
Is the "last last new chunk" the "previous last new chunk"? Like.. the last new chunk of the previous state witness?

Reply:
Yes